engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/function/Function.java

```diff
@@ -201,9 +204,28 @@ static Object doCall(
         Function function,
         Object[] arguments,
         @Cached InteropApplicationNode interopApplicationNode,
-        @CachedLibrary("function") InteropLibrary thisLib) {
-      return interopApplicationNode.execute(
-          function, EnsoContext.get(thisLib).emptyState(), arguments);
+        @CachedLibrary("function") InteropLibrary thisLib,
+        @Cached InlinedBranchProfile panicProfile) {
+      try {
+        return interopApplicationNode.execute(
+            function, EnsoContext.get(thisLib).emptyState(), arguments);
+      } catch (StackOverflowError err) {
```

Comment:
How is JVM even able to call any methods after catching `StackOverflowError`? Is the stack of the current thread that caused the SO temporarily increased? Are you entirely sure that the stacktrace will be printed on SO? Have you tested it?

Reply:
Yes, I did manual testing. `StackOverflowError` isn't caught when it happens, but after the stack is unrolled. The stack when the error is caught [can be seen here](https://github.com/enso-org/enso/pull/11241#discussion_r1787254091) and is actually pretty shallow.